colorizer_data/bin/tfe_open.py

```diff
@@ -15,17 +16,28 @@
 default_directory_port = 6470
 
 
-def get_available_port(default_port):
+def get_available_ports(default_ports: List[int]) -> List[int]:
```

Comment:
smart choice making it recursive! There is a recursion depth limitation with python but we're unlikely to hit it with just few ports so it's totally fine. 
Although another thought I have, return a list of tuples like `[(socket, port), ...]
` instead of just port numbers, then `s.close()` to close the sockets only after the servers running(or on error). This way should be safer because the ports stay reserved until we successfully started the servers on them

Reply:
This is a good idea, thank you, Ruge!